### ISSUE TITLE
Recommends using double-quote when using special characters

### ISSUE THREAD

**Issue by cyberdelia** [rubocop-hq/rubocop#4542]
## Expected behavior

``Style/StringLiterals`` should prevent using single-quote for special symbol: 

```ruby
# bad 
'hello'.split('\t')

# good
'hello'.split("\t")
```

## Actual behavior

Usage of ``'\t' and ``'\n'`` and ``'\''`` are no flagged unlike what the style guide recommends: https://github.com/bbatsov/ruby-style-guide#consistent-string-literals

## Steps to reproduce the problem

```ruby
'hello'.split('\t')
'hello'.split("\t")
```

## RuboCop version

Include the output of `rubocop -V`. Here's an example:

```
$ rubocop -V
0.49.1 (using Parser 2.4.0.0, running on ruby 2.4.1 x86_64-darwin16)
```

**Comment by deivid-rodriguez** [comment]
What if the usage is intentional? Say, `C:\blog\t\my-awesome-topic'.split('\t\\') # => ["C:\\blog", "my-awesome-topic"]`.

Maybe as a disabled by default cop?

**Comment by bbatsov** [comment]
There hasn't been much activity on this ticket and our Core Team is spread too thin on so many tasks, so I'll just close it for the sake of having a cleaner lists of tasks to focus on. We'd gladly review a PR, but it's unlikely that'd we're going to tackle this ourselves in the foreseeable future.
